src/test/java/es/in2/desmos/domain/services/sync/services/DataSyncServiceImplTests.java

```diff
@@ -0,0 +1,148 @@
+                },
+                "description": {
+                    "type": "Property",
+                    "value": "ProductOffering 1 description"
+                }
+            }""";
+
+    @BeforeEach
+    void setup() throws JsonProcessingException {
+    }
+
+    @Test
+    void testVerifyDataIntegrity_Success_FirstEntity() throws JsonProcessingException {
+        JsonNode mockJsonNode = Mockito.mock(JsonNode.class);
+        when(objectMapper.readTree(anyString())).thenReturn(mockJsonNode);
+        when(objectMapper.writeValueAsString(mockJsonNode)).thenReturn(retrievedBrokerEntity);
+        when(auditRecordService.findLatestConsumerPublishedAuditRecord(anyString())).thenReturn(Flux.empty());
+
+
+        StepVerifier.create(dataSyncService.verifyRetrievedEntityData("processId", notification, retrievedBrokerEntity))
+                .assertNext(retrievedBrokerEntity -> {})
+                .verifyComplete();
+    }
+
+    @Test
+    void testVerifyDataIntegrity_Success_SecondEntity() throws JsonProcessingException {
+        JsonNode mockJsonNode = Mockito.mock(JsonNode.class);
+        when(objectMapper.readTree(anyString())).thenReturn(mockJsonNode);
+        when(objectMapper.writeValueAsString(mockJsonNode)).thenReturn(retrievedBrokerEntity);
+        when(auditRecordService.findLatestConsumerPublishedAuditRecord(anyString())).thenReturn(Flux.just(auditRecord));
+
+        StepVerifier.create(dataSyncService.verifyRetrievedEntityData("processId", notification, retrievedBrokerEntity))
+                .assertNext(retrievedBrokerEntity -> {})
+                .verifyComplete();
+
+    }
+
+    @Test
+    void testVerifyDataIntegrity_Failure_HashMismatch() throws JsonProcessingException {
+        JsonNode mockJsonNode = Mockito.mock(JsonNode.class);
+        when(objectMapper.readTree(anyString())).thenReturn(mockJsonNode);
+        when(objectMapper.writeValueAsString(mockJsonNode)).thenReturn(retrievedBrokerEntity);
+        when(auditRecordService.findLatestConsumerPublishedAuditRecord(anyString())).thenReturn(Flux.empty());
+
+
+        StepVerifier.create(dataSyncService.verifyRetrievedEntityData("processId", errorNotification, retrievedBrokerEntity))
+                .expectError(HashLinkException.class)
+                .verify();
+    }
+
+    @Test
+    void testVerifyDataIntegrity_JsonProcessingExceptionError() throws JsonProcessingException {
+        JsonNode mockJsonNode = Mockito.mock(JsonNode.class);
+        when(objectMapper.readTree(anyString())).thenReturn(mockJsonNode);
+        when(objectMapper.writeValueAsString(mockJsonNode)).thenThrow(JsonProcessingException.class);
+        when(auditRecordService.findLatestConsumerPublishedAuditRecord(anyString())).thenReturn(Flux.empty());
+
+
+        StepVerifier.create(dataSyncService.verifyRetrievedEntityData("processId", errorNotification, retrievedBrokerEntity))
+                .expectError(HashLinkException.class)
+                .verify();
+    }
```

Comment:
Test data consistency?

Reply:
That method has the data consistency verification too